### ISSUE TITLE
Add net.Write/ReadEntity that uses EHandles instead of EntIndex

### ISSUE THREAD

**Issue by meepen** [Facepunch/garrysmod-requests#1364]
i believe Entity userdatas are just bound ehandles and can tell validity if the ehandle exists, can you allow us to write and read those ehandles over network?

**Comment by robotboy655** [comment]
What exactly are you asking for? What about `net.Write/ReadEntity` doesn't work for you?

**Comment by Kefta** [comment]
Networking ent indexes is unreliable. If a net message is sent to the server from the client using an ent index that has already been replaced with a new entity the client doesn't yet know about, the server knows none the wiser and uses the wrong entity.

**Comment by thegrb93** [comment]
Typical work-around is https://github.com/thegrb93/StarfallEx/blob/master/lua/entities/starfall_hologram/cl_init.lua#L66

Awful, but it works.

**Comment by thegrb93** [comment]
Yeah, it can mix up entities if player removes/creates stuff quickly.

**Comment by thegrb93** [comment]
I would use DTVars if https://wiki.garrysmod.com/page/Entity/NetworkVarNotify worked clientside

**Comment by thegrb93** [comment]
Looks like https://github.com/Facepunch/garrysmod-requests/issues/324 is almost fixed though.
